qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/PatientSteps.java

```diff
@@ -0,0 +1,260 @@
+	public void createPatient() {
+		testPatient = createTestPatient();
+	}
+	
+	@When("a user clicks on Find Patient Record app from the home page")
+	public void loadFindPatientPage() {
+		findPatientPage = homePage.goToFindPatientRecord();
+	}
+	
+	@And("the system loads find patient page")
+	public void systemLoadsFindPatientPage() {
+		assertTrue(textExists("Find Patient Record"));
+	}
+	
+	@And("a user enters the patient's {string}")
+	public void findPatient(String name) {
+		if ("rightName".equals(name)) {
+			findPatientPage.enterPatient(testPatient.givenName);
+			findPatientPage.waitForPageToLoad();
+		}else {
+			findPatientPage.enterPatient("wrongName");
+		}
+	}
+	
+	@Then("the patient search is {string}")
+	public void systemLoadsPatient(String status) {
+		if ("successful".equals(status)) {
+			assertThat(findPatientPage.getFirstPatientName(), containsString(testPatient.givenName));
+			assertThat(findPatientPage.getFirstPatientName(), containsString(testPatient.middleName));
+			assertThat(findPatientPage.getFirstPatientName(), containsString(testPatient.familyName));
+		} else {
+			assertTrue(textExists("No matching records found"));
+		}
+	}
+	
+	@And("a user searches for an existing patient by name")
+	public void searchPatient() {
+		findPatientPage.enterPatient(testPatient.givenName);
+		firstPatientIdentifier = findPatientPage.getFirstPatientIdentifier();
+		assertNotNull(firstPatientIdentifier);
+	}
+	
+	@And("a user clicks on first patient")
+	public void loadPatientDashboard() {
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+	
+	@Then("the system loads the patient dashboard")
+	public void systemLoadsPatientDashboard() {
+		matchPatientIds(firstPatientIdentifier);
+		dashboardPage.goToHomePage();
+	}
+	
+	@Given("a user clicks on Register a patient App from the home page")
+	public void loadRegisterPatientPage() throws InterruptedException {
+  		registrationPage = homePage.goToRegisterPatientApp();
+	}
+	
+	@When("the system loads Register a patient page")
+	public void systemLoadsRegisterPatientPage() {
+		assertTrue(textExists("Register a patient"));
+	}
+	
+	@And("a user clicks on the Unidentified Patient checkbox")
+	public void clickOnUndefinedPatientCheckbox() {
+  		registrationPage.selectUnidentifiedPatient();
+	}
+	
+	@And("a user selects a gender")
+	public void selectGender() throws InterruptedException {
+  		registrationPage.clickOnGenderLink();
+  		registrationPage.selectPatientGender("Male");
+  		registrationPage.clickOnConfirmSection();
+	}
+	
+	@And("a user clicks Confirm button")
+	public void clickOnConfirmButton() throws InterruptedException {
+  		dashboardPage = registrationPage.confirmPatient();
+	}
+	
+	@Then("the system adds patient record into the patients table")
+	public void systemAddsPatientRecord() {
+		assertThat(dashboardPage.getPatientGivenName(), equalTo("UNKNOWN"));
+		assertThat(dashboardPage.getPatientFamilyName(), equalTo("UNKNOWN"));
```

Comment:
Is patientFamilyName similar to patientGivenName, could you use a different equalTo()

Reply:
> Is patientFamilyName similar to patientGivenName, could you use a different equalTo()

@sherrif10 no they are not similar, only that when an unidentified patient is registered, both the family name and given name are recorded as "UNKNOWN". So the patient will be "UNKNOWN UNKNOWN"